### PR TITLE
Update TCA to 0.58

### PR DESCRIPTION
**Pull request by mbrandonw**
Have another PR coming to update TCA to 1.1 based on this PR.

### CODE REVIEW

File: App/isowords.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -925,7 +927,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "/usr/libexec/PlistBuddy -c \"Set GitSHA '$(git rev-parse --short HEAD)'\" \"${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}\"\n";
+			shellScript = "# TODO: Fix this\n#/usr/libexec/PlistBuddy -c \"Set GitSHA '$(git rev-parse --short HEAD)'\" \"${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}\"\n";
```
Comment (mbrandonw) [inline App/isowords.xcodeproj/project.pbxproj:930]: I need some help with this...

File: App/isowords.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1324,9 +1326,11 @@
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				COPY_PHASE_STRIP = NO;
 				CURRENT_PROJECT_VERSION = 105;
+				DEAD_CODE_STRIPPING = YES;
```
Comment (mbrandonw) [inline App/isowords.xcodeproj/project.pbxproj:1329]: all of this stuff was just Xcode's suggestion to modernize the project.

File: Sources/ActiveGamesFeature/ActiveGamesView.swift [hunk 3]
```diff
@@ -344,9 +344,9 @@ private let relativeDateFormatter = RelativeDateTimeFormatter()
       Preview {
         ScrollView {
           ActiveGamesView(
-            store: .init(
-              initialState: .init(
-                savedGames: .init(
+            store: Store(
+              initialState: ActiveGamesState(
+                savedGames: SavedGamesState(
```
Comment (mbrandonw) [inline Sources/ActiveGamesFeature/ActiveGamesView.swift:349]: Added these when getting some compiler errors, but I don't think they were actually necessary. But also leaving them cause it doesn't hurt.

File: Sources/CombineHelpers/ReplaySubject.swift [hunk 4]
```diff
@@ -44,7 +44,7 @@ public final class ReplaySubject<Output, Failure: Error>: Subject {
     subscription.replay(self.buffer, completion: self.completion)
   }
 
-  private final class Subscription<Output, Failure: Error>: Combine.Subscription {
+  private final class Subscription: Combine.Subscription {
```
Comment (mbrandonw) [inline Sources/CombineHelpers/ReplaySubject.swift:47]: This was a Swift warning and I don't think the shadowed generics were necessary.

File: Sources/DailyChallengeFeature/DailyChallengeView.swift [hunk 5]
```diff
@@ -11,7 +11,7 @@ import SwiftUI
 
 public struct DailyChallengeReducer: ReducerProtocol {
   public struct State: Equatable {
-    public var alert: AlertState<Action>?
+    @PresentationState public var alert: AlertState<Action.Alert>?
```
Comment (mbrandonw) [inline Sources/DailyChallengeFeature/DailyChallengeView.swift:14]: modernized some alerts to get rid of the deprecated `alert(_:dismiss:)` view modifier.

File: Sources/SettingsFeature/Settings.swift [hunk 6]
```diff
@@ -189,116 +193,133 @@ public struct Settings: ReducerProtocol {
...
-
-        case .binding(\.$enableNotifications):
-          guard
-            state.enableNotifications,
-            let userNotificationSettings = state.userNotificationSettings
-          else {
-            // TODO: API request to opt out of all notifications
-            state.enableNotifications = false
-            return .none
+        .onChange(of: \.developer.currentBaseUrl.url) { _, url in
```
Comment (mbrandonw) [inline Sources/SettingsFeature/Settings.swift:204]: The `onChange` stuff for bindings did get pretty gnarly, but not sure if there is a better way to do it.

File: Sources/Styleguide/Padding.swift [hunk 7]
```diff
@@ -33,6 +33,11 @@ private struct ScreenEdgePadding: ViewModifier {
           ?? .grid(20)
       )
 
+#if swift(>=5.9)
+    case .vision:
+      content
+#endif
```
Comment (mbrandonw) [inline Sources/Styleguide/Padding.swift:39]: This was a warning in Xcode 15. Is this the best way to handle it?

File: App/Previews/GameOverPreview/GameOverPreviewApp.swift [hunk 8]
```diff
@@ -36,12 +36,13 @@ extension StoreOf<GameOver> {
...
-      reducer: GameOver()
+      )
+    ) {
+      GameOver()
         .dependency(
           \.apiClient,
-          update(.noop) {
-            $0.override(
+           update(.noop) {
+             $0.override(
```
Comment (stephencelis) [inline App/Previews/GameOverPreview/GameOverPreviewApp.swift:45]:
```suggestion
          update(.noop) {
            $0.override(
```

File: App/Previews/GameOverPreview/GameOverPreviewApp.swift [hunk 9]
```diff
@@ -57,26 +58,26 @@ extension StoreOf<GameOver> {
                   )
                 )
               }
-            )
-          }
+             )
+           }
```
Comment (stephencelis) [inline App/Previews/GameOverPreview/GameOverPreviewApp.swift:62]:
```suggestion
            )
          }
```

File: App/Previews/GameOverPreview/GameOverPreviewApp.swift [hunk 10]
```diff
@@ -57,26 +58,26 @@ extension StoreOf<GameOver> {
...
         )
         .dependency(\.audioPlayer, .noop)
         .dependency(
           \.database,
-          .autoMigratingLive(
+           .autoMigratingLive(
```
Comment (stephencelis) [inline App/Previews/GameOverPreview/GameOverPreviewApp.swift:67]:
```suggestion
          .autoMigratingLive(
```

File: App/Previews/GameOverPreview/GameOverPreviewApp.swift [hunk 11]
```diff
@@ -57,26 +58,26 @@ extension StoreOf<GameOver> {
...
             path: FileManager.default
               .urls(for: .documentDirectory, in: .userDomainMask)
               .first!
               .appendingPathComponent("co.pointfree.Isowords")
               .appendingPathComponent("Isowords.sqlite3")
-          )
+           )
```
Comment (stephencelis) [inline App/Previews/GameOverPreview/GameOverPreviewApp.swift:73]:
```suggestion
          )
```

File: App/Previews/HomeFeaturePreview/HomeFeaturePreviewApp.swift [hunk 12]
```diff
@@ -18,13 +18,14 @@ struct HomeFeaturePreviewApp: App {
...
+          ) {
+            Home()
               .dependency(
                 \.apiClient,
-                update(.noop) {
-                  $0.authenticate = { _ in .init(appleReceipt: nil, player: .blob) }
-                  $0.override(
+                 update(.noop) {
+                   $0.authenticate = { _ in .init(appleReceipt: nil, player: .blob) }
+                   $0.override(
```
Comment (stephencelis) [inline App/Previews/HomeFeaturePreview/HomeFeaturePreviewApp.swift:28]:
```suggestion
                update(.noop) {
                  $0.authenticate = { _ in .init(appleReceipt: nil, player: .blob) }
                  $0.override(
```

File: App/Previews/HomeFeaturePreview/HomeFeaturePreviewApp.swift [hunk 13]
```diff
@@ -43,8 +44,8 @@ struct HomeFeaturePreviewApp: App {
                         )
                       ])
                     }
-                  )
-                }
+                   )
+                 }
```
Comment (stephencelis) [inline App/Previews/HomeFeaturePreview/HomeFeaturePreviewApp.swift:48]:
```suggestion
                  )
                }
```

File: App/Previews/TrailerPreview/TrailerPreviewApp.swift [hunk 14]
```diff
@@ -16,34 +16,35 @@ struct TrailerPreviewApp: App {
...
         store: .init(
-          initialState: Trailer.State(),
-          reducer: Trailer()
+          initialState: Trailer.State()
+        ) {
+          Trailer()
             .dependency(
               \.audioPlayer,
-              .live(bundles: [
+               .live(bundles: [
```
Comment (stephencelis) [inline App/Previews/TrailerPreview/TrailerPreviewApp.swift:24]:
```suggestion
              .live(bundles: [
```

File: App/Previews/TrailerPreview/TrailerPreviewApp.swift [hunk 15]
```diff
@@ -16,34 +16,35 @@ struct TrailerPreviewApp: App {
...
                 AppAudioLibrary.bundle,
                 AppClipAudioLibrary.bundle,
-              ])
+               ])
```
Comment (stephencelis) [inline App/Previews/TrailerPreview/TrailerPreviewApp.swift:27]:
```suggestion
              ])
```

File: App/Previews/TrailerPreview/TrailerPreviewApp.swift [hunk 16]
```diff
@@ -16,34 +16,35 @@ struct TrailerPreviewApp: App {
...
             )
             .dependency(
               \.dictionary,
-              .init(
+               .init(
```
Comment (stephencelis) [inline App/Previews/TrailerPreview/TrailerPreviewApp.swift:31]:
```suggestion
              .init(
```

File: App/Previews/TrailerPreview/TrailerPreviewApp.swift [hunk 17]
```diff
@@ -16,34 +16,35 @@ struct TrailerPreviewApp: App {
...
                 contains: { string, _ in
                   [
                     "SAY", "HELLO", "TO", "ISOWORDS",
                     "A", "NEW", "WORD", "SEARCH", "GAME",
                     "FOR", "YOUR", "PHONE",
                     "COMING", "NEXT", "YEAR",
                   ]
-                  .contains(string.uppercased())
+                    .contains(string.uppercased())
```
Comment (stephencelis) [inline App/Previews/TrailerPreview/TrailerPreviewApp.swift:39]:
```suggestion
                  .contains(string.uppercased())
```

File: App/Previews/TrailerPreview/TrailerPreviewApp.swift [hunk 18]
```diff
@@ -16,34 +16,35 @@ struct TrailerPreviewApp: App {
...
                 },
                 load: { _ in true },
                 lookup: { _, _ in nil },
                 randomCubes: { _ in .mock },
                 unload: { _ in }
-              )
+               )
```
Comment (stephencelis) [inline App/Previews/TrailerPreview/TrailerPreviewApp.swift:45]:
```suggestion
              )
```

File: Sources/ChangelogFeature/ChangelogView.swift [hunk 19]
```diff
@@ -173,13 +175,14 @@ public struct ChangelogView: View {
...
+          ) {
+            ChangelogReducer()
               .dependency(
                 \.apiClient,
-                {
-                  var apiClient = ApiClient.noop
-                  apiClient.override(
+                 {
+                   var apiClient = ApiClient.noop
+                   apiClient.override(
```
Comment (stephencelis) [inline Sources/ChangelogFeature/ChangelogView.swift:185]:
```suggestion
                {
                  var apiClient = ApiClient.noop
                  apiClient.override(
```

File: Sources/ChangelogFeature/ChangelogView.swift [hunk 20]
```diff
@@ -194,14 +197,14 @@ public struct ChangelogView: View {
                         }
                       )
                     }
-                  )
-                  return apiClient
-                }()
+                   )
+                   return apiClient
+                 }()
```
Comment (stephencelis) [inline Sources/ChangelogFeature/ChangelogView.swift:202]:
```suggestion
                  )
                  return apiClient
                }()
```

File: Sources/SettingsFeature/Settings.swift (hunk 6 above)
Comment (stephencelis) [reply Sources/SettingsFeature/Settings.swift:203]: On the other hand, we've introduced isowords bugs in the past from nested binding state...

File: Sources/TrailerFeature/Trailer.swift [hunk 21]
```diff
@@ -90,7 +90,7 @@ public struct Trailer: ReducerProtocol {
           try await self.mainQueue.sleep(for: firstWordDelay)
 
           // Play each word
-          for (wordIndex, word) in replayableWords.enumerated() {
+          for (_, word) in replayableWords.enumerated() {
```
Comment (stephencelis) [inline Sources/TrailerFeature/Trailer.swift:93]: Can this just be:
```suggestion
          for word in replayableWords {
```

**Review by stephencelis — APPROVED** [review]

**Pull request merged**